src/strategy.py

```diff
@@ -115,13 +114,16 @@ def step(self, event_list=[]):
                     assert vehicle.battery.soc >= (1-self.margin)*vehicle.desired_soc - self.EPS, (
                         "{}: Vehicle {} is below desired SOC ({} < {})".format(
                             ev.start_time.isoformat(), ev.vehicle_id,
-                            vehicle.battery.soc, vehicle.desired_soc))
+                            vehicle.battery.soc, vehicle.desired_soc - self.margin))
```

Comment:
No. The error message should display the real desired SoC. Internally, the margin is applied, but if the desired SoC is not reached with the margin, it is definitely not reached without it.